explainerdashboard/explainer_methods.py

```diff
@@ -1791,3 +1796,25 @@ def get_xgboost_preds_df(xgbmodel, X_row, pos_label=1):
             0, "pred_proba"
         ]
     return xgboost_preds_df
+
+
+def check_dtype_of(
+    features: List[str],
+    df_target: pd.DataFrame,
+    df_origin: pd.DataFrame
+):
+    """
+        Check if the features in the df_target has the same dtype in df_origin,
+        if not, it fixes the dtype in df_target
+    :param features:
+    :param df_target:
+    :param df_origin:
+    """
+    # Verify if the features has the same dtype as the X sample
+    if (
+            df_target is not None and
+            not df_target[features].dtypes.eq(df_origin[features].dtypes).all()
+    ):
+        df_target[features] = df_target[features].astype(
```

Comment:
in general not a fan of these functions that modify in place. Could you rewrite it such that it returns the transformed df instead? Then maybe call it `adjust_dtypes_to_match_df(...)` or something?

Calling something `check_dtype_of` when it actually modifies one of the arguments is confusing.